bbconf/config_parser/bedbaseconfig.py

```diff
@@ -0,0 +1,338 @@
+        return self._t2bsi
+
+    @property
+    def r2v(self) -> Region2VecExModel:
+        """
+        Get region2vec object
+
+        :return: region2vec object
+        """
+        return self._r2v
+
+    @property
+    def qdrant_engine(self) -> QdrantBackend:
+        """
+        Get qdrant engine
+
+        :return: qdrant engine
+        """
+        return self._qdrant_engine
+
+    @property
+    def phc(self) -> PEPHubClient:
+        """
+        Get PEPHub client
+
+        :return: PEPHub client
+        """
+        return self._phc
+
+    @property
+    def boto3_client(self) -> boto3.client:
+        """
+        Get boto3 client
+
+        :return: boto3 client
+        """
+        return self._boto3_client
+
+    def _init_db_engine(self) -> BaseEngine:
+        return BaseEngine(
+            host=self._config.database.host,
+            port=self._config.database.port,
+            database=self._config.database.database,
+            user=self._config.database.user,
+            password=self._config.database.password,
+            drivername="postgresql+psycopg",
+        )
+
+    def _init_qdrant_backend(self) -> QdrantBackend:
+        """
+        Create qdrant client object using credentials provided in config file
+
+        :return: QdrantClient
+        """
+        try:
+            return QdrantBackend(
+                collection=self._config.qdrant.collection,
+                qdrant_host=self._config.qdrant.host,
+                qdrant_port=self._config.qdrant.port,
+                qdrant_api_key=self._config.qdrant.api_key,
+            )
+        except qdrant_client.http.exceptions.ResponseHandlingException as err:
+            _LOGGER.error(f"error in Connection to qdrant! skipping... Error: {err}")
+            warnings.warn(
+                f"error in Connection to qdrant! skipping... Error: {err}", UserWarning
+            )
+
+    def _init_t2bsi_object(self) -> Union[text2bednn.Text2BEDSearchInterface, None]:
+        """
+        Create Text 2 BED search interface and return this object
+
+        :return: Text2BEDSearchInterface object
+        """
+
+        try:
+            return text2bednn.Text2BEDSearchInterface(
+                nl2vec_model=FlagEmbedding(model_name=self._config.path.text2vec),
+                vec2vec_model=self._config.path.vec2vec,
+                search_backend=self.qdrant_engine,
+            )
+        except Exception as e:
+            _LOGGER.error("Error in creating Text2BEDSearchInterface object: " + str(e))
+            warnings.warn(
+                "Error in creating Text2BEDSearchInterface object: " + str(e),
+                UserWarning,
+            )
+            return None
+
+    @staticmethod
+    def _init_pephubclient() -> Union[PEPHubClient, None]:
+        """
+        Create Pephub client object using credentials provided in config file
+
+        :return: PephubClient
+        """
+        try:
+            return PEPHubClient()
```

Comment:
Blank?

Reply:
yes, because for now, we can't provide any password